app/javascript/widget/App.vue

```diff
@@ -184,6 +188,11 @@ export default {
           this.showUnreadView = true;
         } else if (message.event === 'unset-unread-view') {
           this.showUnreadView = false;
+        } else if (message.event === 'change-url') {
+          startCampaigns({
+            allCampaigns: this.campaigns,
+            currentURL: message.url,
+          });
```

Comment:
I would suggest this also to be an action in the campaign module. Otherwise, changes in campaigns object would re-render the entire tree.